diff -Nur a/plat/nuvoton/npcm845x/npcm845x_psci.c b/plat/nuvoton/npcm845x/npcm845x_psci.c
--- a/plat/nuvoton/npcm845x/npcm845x_psci.c	2022-03-10 14:44:12.433108700 +0200
+++ b/plat/nuvoton/npcm845x/npcm845x_psci.c	2022-03-20 13:39:47.615868654 +0200
@@ -256,16 +256,28 @@
 
 void __dead2 npcm845x_system_reset(void)
 {
+	unsigned long RESET_BASE_ADDR;
 	NOTICE("%s() nuvoton_psci\n", __func__);
 	console_flush();
 
 	dsbsy();
 	isb();
 	// In future - support all reset types. For now, SW1 reset
-    arbel_handle_reset(RESET_TYPE_SW1);
-	for (;;) {
-		wfi();
-	}
+
+	RESET_BASE_ADDR = (uintptr_t)0xF0801000;
+	mmio_write_32(RESET_BASE_ADDR + 0x14, 0x8);
+	dsb();
+	/*
+	 * Maybe the HW needs some times to reset the system,
+	 * so we do not hope the core to excute valid codes.
+	 */
+	while (1)
+		;
+
+	//arbel_handle_reset(RESET_TYPE_SW1);
+	//for (;;) {
+	//	wfi();
+	//}
 }
 
 int npcm845x_validate_power_state(unsigned int power_state,
