From b608c1499087c5e0675f8fd9c22cf172273a17a3 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Thu, 10 Feb 2022 10:36:51 +0800
Subject: [PATCH] Fix wrong type of TRANSACTION_ID metadata

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 phosphor-logging/log.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/phosphor-logging/log.hpp b/phosphor-logging/log.hpp
index 14c0faa7..4c826ac0 100644
--- a/phosphor-logging/log.hpp
+++ b/phosphor-logging/log.hpp
@@ -163,7 +163,7 @@ void log(Msg msg, Entry... e)
     constexpr const char* msg_str = "MESSAGE=%s";
     const auto msg_tuple = std::make_tuple(msg_str, std::forward<Msg>(msg));
 
-    constexpr auto transactionStr = "TRANSACTION_ID=%lld";
+    constexpr auto transactionStr = "TRANSACTION_ID=%llu";
     auto transactionId = sdbusplus::server::transaction::get_id();
 
     auto log_tuple = std::tuple_cat(details::prio<L>(), msg_tuple,
-- 
2.17.1

