diff --git a/include/configs/arbel.h b/include/configs/arbel.h
index b4ebc54641..ab7f308b61 100644
--- a/include/configs/arbel.h
+++ b/include/configs/arbel.h
@@ -38,7 +38,7 @@
 
 /* Default environemnt variables */
 #define CONFIG_BOOTCOMMAND "run common_bootargs; run romboot"
-#define CONFIG_EXTRA_ENV_SETTINGS   "uimage_flash_addr=80200000\0"   \
+#define CONFIG_EXTRA_ENV_SETTINGS   "uimage_flash_addr=80400000\0"   \
 		"stdin=serial\0"   \
 		"stdout=serial\0"   \
 		"stderr=serial\0"    \
-- 
2.17.1

