From 0c66a7f348692507b5e4e0683a557c768fca1f67 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Tue, 12 Jul 2022 15:17:49 +0800
Subject: [PATCH] change the cpu sensor name from CPU to CPU_Utilization

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 healthMonitor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/healthMonitor.cpp b/healthMonitor.cpp
index be62c3a..9d56e1d 100644
--- a/healthMonitor.cpp
+++ b/healthMonitor.cpp
@@ -235,7 +235,7 @@ constexpr auto storage = "Storage";
 constexpr auto inode = "Inode";
 /** Map of read function for each health sensors supported */
 const std::map<std::string, std::function<double(std::string path)>>
-    readSensors = {{"CPU", readCPUUtilization},
+    readSensors = {{"CPU_Utilization", readCPUUtilization},
                    {"Memory", readMemoryUtilization},
                    {storage, readStorageUtilization},
                    {inode, readInodeUtilization}};
-- 
2.34.1

