From 8e9cc4221d998cf9840696920d8bb76b1f53000d Mon Sep 17 00:00:00 2001
From: Tim Lee <timlee660101@gmail.com>
Date: Thu, 28 Oct 2021 16:59:47 +0800
Subject: [PATCH 5/5] bmcweb chassis add indicatorLED support

Signed-off-by: Tim Lee <timlee660101@gmail.com>
---
 redfish-core/lib/chassis.hpp | 106 +++++++++++++++++++++++++++++++++--
 1 file changed, 101 insertions(+), 5 deletions(-)

diff --git a/redfish-core/lib/chassis.hpp b/redfish-core/lib/chassis.hpp
index 5a837936e..4bdf908a7 100644
--- a/redfish-core/lib/chassis.hpp
+++ b/redfish-core/lib/chassis.hpp
@@ -119,6 +119,73 @@ inline void getIntrusionByService(std::shared_ptr<bmcweb::AsyncResp> aResp,
         "xyz.openbmc_project.Chassis.Intrusion", "Status");
 }
 
+/**
+ * Retrieves physical Led properties over dbus
+ */
+void getPhysicalLedData(std::shared_ptr<bmcweb::AsyncResp> aResp, std::string path)
+{
+    std::size_t found = path.rfind('/');
+    std::string led_name;
+
+    if (found != std::string::npos)
+    {
+        led_name = path.substr(found + 1);
+    }
+    else
+    {
+        return;
+    }
+
+    std::string obj = "xyz.openbmc_project.LED.Controller." + led_name;
+
+    crow::connections::systemBus->async_method_call(
+        [aResp](const boost::system::error_code ec,
+                                        const PropertiesType &properties) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                messages::internalError(aResp->res);
+                return;
+            }
+            BMCWEB_LOG_DEBUG << "Got " << properties.size()
+                             << " led properties.";
+            std::string output;
+            for (const auto &property : properties)
+            {
+                if (property.first == "State")
+                {
+                    const std::string *s =
+                        std::get_if<std::string>(&property.second);
+                    if (nullptr != s)
+                    {
+                        BMCWEB_LOG_DEBUG << "Identify Led State: " << *s;
+                        const auto pos = s->rfind('.');
+                        if (pos != std::string::npos)
+                        {
+                            auto led = s->substr(pos + 1);
+                            for (const std::pair<const char *, const char *>
+                                     &p :
+                                 std::array<
+                                     std::pair<const char *, const char *>, 3>{
+                                     {{"On", "Lit"},
+                                      {"Blink", "Blinking"},
+                                      {"Off", "Off"}}})
+                            {
+                                if (led == p.first)
+                                {
+                                    aResp->res.jsonValue["IndicatorLED"] = p.second;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        obj, path,
+        "org.freedesktop.DBus.Properties", "GetAll",
+        "xyz.openbmc_project.Led.Physical");
+}
+
 /**
  * Retrieves physical security properties over dbus
  */
@@ -225,8 +292,39 @@ inline void requestRoutesChassis(App& app)
                         auto health =
                             std::make_shared<HealthPopulate>(asyncResp);
 
+                        // Get the list of all sensors for this Chassis element
+                        std::vector<std::string> sensorPaths = {
+                            "/all_sensors",
+                            "/sensors"
+                        };
+
+                        for (const std::string& sensorPath : sensorPaths)
+                        {
+                            crow::connections::systemBus->async_method_call(
+                                [health](const boost::system::error_code ec,
+                                        std::variant<std::vector<std::string>> &resp) {
+                                    if (ec)
+                                    {
+                                        return; // no sensors = no failures
+                                    }
+                                    std::vector<std::string> *data =
+                                        std::get_if<std::vector<std::string>>(&resp);
+                                    if (data == nullptr)
+                                    {
+                                        return;
+                                    }
+                                    health->inventory = std::move(*data);
+                                },
+                                "xyz.openbmc_project.ObjectMapper",
+                                path + sensorPath,
+                                "org.freedesktop.DBus.Properties", "Get",
+                                "xyz.openbmc_project.Association", "endpoints");
+                        }
+
+                        health->populate();
+
                         crow::connections::systemBus->async_method_call(
-                            [health](
+                            [asyncResp](
                                 const boost::system::error_code ec2,
                                 std::variant<std::vector<std::string>>& resp) {
                                 if (ec2)
@@ -240,15 +338,13 @@ inline void requestRoutesChassis(App& app)
                                 {
                                     return;
                                 }
-                                health->inventory = std::move(*data);
+                                getPhysicalLedData(asyncResp, std::move((*data)[0]));
                             },
                             "xyz.openbmc_project.ObjectMapper",
-                            path + "/all_sensors",
+                            path + "/leds",
                             "org.freedesktop.DBus.Properties", "Get",
                             "xyz.openbmc_project.Association", "endpoints");
 
-                        health->populate();
-
                         if (connectionNames.size() < 1)
                         {
                             BMCWEB_LOG_ERROR << "Got 0 Connection names";
-- 
2.17.1

