From 4c24f14dc44b1f79c884b22dc1fb8e33a3256280 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Mon, 11 Jul 2022 17:42:10 +0800
Subject: [PATCH] dbus-sdr: do not replace '_' for sensor name

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 dbus-sdr/sensorcommands.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbus-sdr/sensorcommands.cpp b/dbus-sdr/sensorcommands.cpp
index f0f99a36..fb3018dc 100644
--- a/dbus-sdr/sensorcommands.cpp
+++ b/dbus-sdr/sensorcommands.cpp
@@ -432,7 +432,7 @@ std::string parseSdrIdFromPath(const std::string& path)
         }
         name.resize(FULL_RECORD_ID_STR_MAX_LENGTH);
     }
-    std::replace(name.begin(), name.end(), '_', ' ');
+    //std::replace(name.begin(), name.end(), '_', ' ');
     return name;
 }
 
-- 
2.34.1

