From 9c9515f5bb451ccb9506b0ff796996b5fc6c71e1 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Fri, 22 Jul 2022 12:51:42 +0800
Subject: [PATCH] add post complete target

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 src/power_control.cpp | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index bada386..e0e4515 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -400,6 +400,24 @@ static constexpr auto systemdPath = "/org/freedesktop/systemd1";
 static constexpr auto systemdInterface = "org.freedesktop.systemd1.Manager";
 static constexpr auto chassisOnTargetName = "obmc-chassis-poweron.target";
 static constexpr auto chassisOffTargetName = "obmc-chassis-poweroff.target";
+static constexpr auto postCompleteTargetName = "obmc-post-complete.target";
+
+void HostPostCompleted()
+{
+    conn->async_method_call(
+        [](boost::system::error_code ec) {
+            if (ec)
+            {
+                std::string errMsg =
+                    "Failed to call chassisPowerOn: " + ec.message();
+                phosphor::logging::log<phosphor::logging::level::ERR>(
+                    errMsg.c_str());
+            }
+        },
+        systemdBusname, systemdPath, systemdInterface, "StartUnit",
+        postCompleteTargetName, "replace");
+    return;
+}
 
 void chassisPowerOn()
 {
@@ -2315,6 +2333,7 @@ static void postCompleteHandler(bool state)
     {
         sendPowerControlEvent(Event::postCompleteAssert);
         osIface->set_property("OperatingSystemState", std::string("Standby"));
+        HostPostCompleted();
     }
     else
     {
-- 
2.34.1

